service_discovery_manager/Tools/weightFinder/renderer/server.py

```diff
@@ -0,0 +1,214 @@
+#
+# Sets renderer configs on Ubuntu 16.04 
+# cpufreq-info, cpulimit command must be avaliable to run the Automation code
+# edit GRUB_CMDLINE_LINUX_DEFAULT in /etc/default/grub and add 'intel_pstate=disable'
+# 'sudo update-grub' must be performed after editing GRUB_CMDLINE_LINUX_DEFAULT
+# and must reboot after setting new grub
+# Prior to run the code, chrome renderer should be running
+#
+# This auto-test code is only tested on Ubuntu 16.04
+# Use it at your own risk and feel free to modify the code
+#
+
+import socket, json, atexit, os, sys
+from subprocess import Popen, PIPE
+from subprocess import check_output
+
+global g_port
+
+g_availableFreq = [3200, 3000, 2900, 2700, 2500, 2300, 2200, 2000, 1800, 1700, 1500, 1300, 1100, 1000, 800] # available MHz by cpufrq-info cmd
+g_procName = "chrome"
+g_cgroupDir = "/dev/cgroups"
+g_groupMemLimitDir = g_cgroupDir+"/limitGroup"
+
+g_numCores = 1 # modifies at init
+g_chromePIDs = []
+
+def exit_handler():
+    cmd = 'tc qdisc del dev enp4s0 root'
+    os.system(cmd)
+    cmd = 'killall cpulimit'
+    os.system(cmd)
+    return
+
+def get_pid(name):
+    g_chromePIDs =  map(int,check_output(["pidof",name]).split())
+    return map(int,check_output(["pidof",name]).split())
+    #print g_chromePIDs;
+
+def setNetworkConfig(latency, packetDropRate):
+    cmd = 'tc qdisc replace dev enp4s0 root handle 1:0 netem loss '+str(packetDropRate)+'% delay ' + str(latency) +'ms'
+    os.system(cmd)
+    return
+
+'''
+def setNetworkLatency(latency):
+    print ("Set Latnecy: "+str(latency))
+    cmd = 'tc qdisc replace dev enp4s0 root handle 1:0 netem delay ' + str(latency)+'ms'
+    print cmd
+    os.system(cmd)
+    return
+
+def setNetworkBandwidth(packetDropRate): #since only egress can be controlled and ingress cannot be controled, we control the bandwidth with packet drop rate.
+    print "Set packetDropRate: " + str(packetDropRate)
+    cmd = 'tc qdisc replace dev enp4s0 parent 1:0 handle 10: netem loss ' + str(packetDropRate)+'%'
+    print cmd
+    os.system(cmd)
+    return
+'''
+
+def setCPUUtilization(utilization):
+    utilization = (100 - utilization)*g_numCores # limit cpu usage (Current CPU utilization 30% = my proc can use maximum 70% of the CPU)
```

Comment:
Need a space among operand and operator.

Reply:
Done